pkg/models/observer_buffer.go

```diff
@@ -55,6 +55,25 @@ type ObserverBuffer struct {
 	MaxE2ELatency       time.Duration
 	MinE2ELatency       time.Duration
 	SumE2ELatency       time.Duration
+
+	InvalidErrors map[SanitisedErrorMetadata]int
+	FailedErrors  map[SanitisedErrorMetadata]int
+}
+
+func (b *ObserverBuffer) appendInvalidError(msgs []*Message) {
+	for _, msg := range msgs {
+		if sem, ok := msg.GetError().(SanitisedErrorMetadata); ok {
+			b.InvalidErrors[sem] = b.InvalidErrors[sem] + 1
+		}
+	}
+}
+
+func (b *ObserverBuffer) appendFailedError(msgs []*Message) {
+	for _, msg := range msgs {
+		if sem, ok := msg.GetError().(SanitisedErrorMetadata); ok {
+			b.FailedErrors[sem] = b.FailedErrors[sem] + 1
+		}
+	}
```

Comment:
There is also `func (b *ObserverBuffer) String() string` function which is sometimes helpful when debugging. 

Is there any neat way we could incorporate these two in there? cc @colmsnowplow question is whether we want to

Reply:
I mean this function https://github.com/snowplow/snowbridge/blob/a075d296a0f802b97a414d188596f3543aac8432/pkg/models/observer_buffer.go#L234

Can we(should we) somehow add info about new `InvalidErrors` and `FailedErrors` there?